src/katgpucbf/xbgpu/engine.py

```diff
@@ -318,14 +749,14 @@ def __init__(
         # most tests cases up until now. If the pipeline starts bottlenecking,
         # then maybe look at increasing these values.
         n_rx_items = 3  # Too high means too much GPU memory gets allocated
-        n_tx_items = 2
 
         # Multiply this _step by 2 to account for dropping half of the
         # spectrum due to symmetric properties of the Fourier Transform. While
         # we can workout the timestamp_step from other parameters that
         # configure the receiver, we pass it as a seperate argument to the
         # reciever for cases where the n_channels_per_stream changes across
         # streams (likely for non-power-of-two array sizes).
+        # NOTE: Should be the same for both X- and B-engines, right?
```

Comment:
The thought occurs to me to wonder whether this comment actually relates to the code anymore. It may be worth digging back into the history. There's no 2 here, was there at one point?

Reply:
And yes, n_channels_per_stream will be the same for all streams because it's determined by the input that the engine subscribes to (and non-power-of-two array sizes have nothing to do with it).